ui/app/components/job-status/panel.js

```diff
@@ -0,0 +1,81 @@
+// @ts-check
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+
+export default class JobStatusPanelComponent extends Component {
+  // Build note: allocTypes order matters! We will fill up to 100% of totalAllocs in this order.
+  allocTypes = [
+    'running',
+    'pending',
+    'failed',
+    // 'unknown',
+    // 'lost',
+    // 'queued',
+    // 'complete',
+    'unplaced',
+  ].map((type) => {
+    return {
+      label: type,
+      property: `${type}Allocs`,
+    };
+  });
+
+  get allocBlocks() {
+    let availableSlotsToFill = this.totalAllocs;
+    // Only fill up to 100% of totalAllocs. Once we've filled up, we can stop counting.
+    let allocationsOfShowableType = this.allocTypes.reduce((blocks, type) => {
+      const jobAllocsOfType = this.args.job.allocations.filterBy(
+        'clientStatus',
+        type.label
+      );
+      if (availableSlotsToFill > 0) {
+        blocks[type.label] = Array(
+          Math.min(availableSlotsToFill, jobAllocsOfType.length)
+        )
+          .fill()
+          .map((_, i) => {
+            return jobAllocsOfType[i];
+          });
+        availableSlotsToFill -= blocks[type.label].length;
+      } else {
+        blocks[type.label] = [];
+      }
+      return blocks;
+    }, {});
+    if (availableSlotsToFill > 0) {
+      allocationsOfShowableType['unplaced'] = Array(availableSlotsToFill)
+        .fill()
+        .map(() => {
+          return { clientStatus: 'unplaced' };
+        });
+    }
+    return allocationsOfShowableType;
```

Comment:
`thought`: Maybe this component deserves an integration test? This component does a decent amount of derived state computation and the associated template has 4 directives and a large number of conditional CSS styling.

Reply:
Hey @ChaiWithJai , about 40% of this PR is in new tests: https://github.com/hashicorp/nomad/pull/16134/files#diff-b60948b235093e292a57d1a3ae77e92f8e955a27d44cfb8914c3e28be0c21d44 which include the resultant blocks from the code above. It's set up as an acceptance test, but could be converted to an integration test if we wanted to run this in a closer-to-the-logic kind of way. Was there anything in particular that you're worried about test coverage for?